src/renderer/components/uielements/drag/Drag.tsx

```diff
@@ -124,32 +101,26 @@ const Drag: React.FC<Props> = ({
       setOverlap(false)
       setDisabled(true)
       onConfirm()
-    } else {
-      handleReset()
     }
+    handleReset()
   }
 
   const dragHandlers = {
     onStart: handleDragStart,
     onDrag: handleDragging,
     onStop: handleDragStop
   }
-
   return (
     <div className={`drag-wrapper ${className}`}>
-      <DragWrapper overlap={overlap} success={success} missed={missed} dragging={dragging} {...rest}>
+      <Styled.DragContainer overlap={overlap} success={success} missed={missed} dragging={dragging} {...rest}>
         <Draggable position={pos} axis="x" bounds={dragBounds} {...dragHandlers}>
-          <AssetIcon
-            onMouseEnter={handleFocus}
-            onMouseLeave={handleBlur}
-            className="source-asset"
-            asset={source}
-            size="small"
-          />
+          <div className="source-asset">
+            <Styled.AssetIcon asset={source} size="small" />
+          </div>
         </Draggable>
-        {title && <TitleLabel color="input">{title}</TitleLabel>}
+        {title && <Styled.TitleLabel color="input">{title}</Styled.TitleLabel>}
         <AssetIcon className="target-asset" asset={target} size="small" />
```

Comment:
[nitpick] Avoid to use `className` by defining a `styled-component` in `Drag.style.ts`. _I know that's not a code you have written..._

Reply:
you are totally right but this `className` is used from the outside to set additional styles for this component. i did not want to rewrite this to use custom render prop.
but yes. u r totally right. its not clear logic here

Reply:
To avoid same mistakes we have made in BEPSwap, we should try to avoid such things if we see such things...